### PR TITLE
Add images to tvdb series search results

### LINKED ISSUE #4368: Artwork from TheTVDB isn't downloading
**Describe the bug**
Download artwork (poster, backgrounds etc) is not working for TheTVDB.

**System (please complete the following information):**
 - OS: Linux (Synology)
 - Virtualization: Docker
 - Jellyfin Version: 10.6.4
 - Installed Plugins: None
 - Reverse Proxy: NGINX
 - Base URL: https://jellyfin.mydomain.com
 - Networking: Bridge (also tried macvlan)
 - Storage: local

**To Reproduce**
1. Add TV Serie library
2. Do a scan

**Expected behavior**
Should download poster art, season art etc from TheTVDB

**Logs**
https://pastebin.com/2qR77W15

### PR DESCRIPTION
**Pull request by cvium**
**Changes**
Moved some stuff around to make it easier to set the image url from tvdbseriesprovider

**Issues**
Fixes #4368

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs [hunk 1]
```diff
@@ -75,27 +58,98 @@ public async Task<MetadataResult<Series>> GetMetadata(SeriesInfo itemId, Cancell
...
+                var zap2ItId = seriesInfo.GetProviderId(MetadataProvider.Zap2It);
+                if (!string.IsNullOrEmpty(zap2ItId))
+                {
+                    tvdbId = await GetSeriesByRemoteId(
+                        zap2ItId,
+                        MetadataProvider.Zap2It.ToString(),
+                        seriesInfo.MetadataLanguage,
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
```
Comment (BaronGreenback) [inline MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs:108]: Can't the 2nd if be moved inside the first?
```suggestion

            if (string.IsNullOrEmpty(tvdbId))
            {
                var zap2ItId = seriesInfo.GetProviderId(MetadataProvider.Zap2It);
                if (!string.IsNullOrEmpty(zap2ItId))
                {
                    tvdbId = await GetSeriesByRemoteId(
                        zap2ItId,
                        MetadataProvider.Zap2It.ToString(),
                        seriesInfo.MetadataLanguage,
                        cancellationToken).ConfigureAwait(false);
                }
            }
        }
```

**Review by BaronGreenback — CHANGES_REQUESTED** [review]

File: MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs (hunk 1 above)
Comment (cvium) [reply MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs:108]: There's no benefit to it. I moved it outside the first if to reduce nesting.

File: MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs [hunk 2]
```diff
@@ -75,27 +58,98 @@ public async Task<MetadataResult<Series>> GetMetadata(SeriesInfo itemId, Cancell
...
+                return Array.Empty<RemoteSearchResult>();
+            }
+        }
+
+        private RemoteSearchResult MapSeriesToRemoteSearchResult(TvDbSharper.Dto.Series series)
+        {
+            var remoteResult = new RemoteSearchResult
+            {
+                Name = series.SeriesName,
+                Overview = (series.Overview ?? string.Empty).Trim(),
```
Comment (BaronGreenback) [inline MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs:130]: Wouldn't this be better as it wouldnt result in the possibility that string.Empty got trimmed?

```suggestion
                Overview = series?.Overview.Trim() ?? string.Empty,
```

File: MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs [hunk 3]
```diff
@@ -75,27 +58,98 @@ public async Task<MetadataResult<Series>> GetMetadata(SeriesInfo itemId, Cancell
...
+                SearchProviderName = Name,
+                ImageUrl = TvdbUtils.BannerUrl + series.Poster
+            };
+
+            if (DateTime.TryParse(series.FirstAired, out var date))
+            {
+                // dates from tvdb are either EST or capital of primary airing country, fuck that noise
```
Comment (BaronGreenback) [inline MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs:137]:
```suggestion
                // Dates from tvdb are either EST or capital of primary airing country.
```

**Review by BaronGreenback — CHANGES_REQUESTED** [review]

**Comment by crobibero** [comment]
This should be implemented in https://github.com/jellyfin/jellyfin-plugin-tvdb instead of on server.

**Pull request closed without merging**